src/Build.UnitTests/BackEnd/BuildManager_Tests.cs

```diff
@@ -4093,6 +4093,32 @@ public void IdenticalSubmissionsShouldCompleteAndNotHangTheBuildOnMissingTargetE
             }
         }
 
+        [Fact]
+        public void BuildWithZeroConnectionTimeout()
+        {
+            string contents = CleanupFileContents(@"
```

Comment:
I'm a little confused why this works. If _env is reused, shouldn't this affect other tests? And why doesn't this build in-proc, avoiding the node connection timeout? (I saw that _env had set MSBUILDINPROCENVCHECK to 1, but I didn't really understand what that was doing.)

Reply:
> If _env is reused, shouldn't this affect other tests?

No, `_env` is an instance field on the test and a new instance is created for each test case. I believe that this is a relevant reference: https://xunit.net/docs/shared-context

> And why doesn't this build in-proc, avoiding the node connection timeout?

It does build in-proc. The bug actually reproduces with single-proc builds. It threw me off as well until I realized that `NodeProviderInProc` starting a new thread in the current process and waiting for it to get initialized is where the env var is used.

Reply:
Neat, thanks!